optimize/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+extern crate jni;
+
+use jni::objects::JClass;
+use jni::objects::JList;
+use jni::objects::JString;
+use jni::JNIEnv;
+use aho_corasick::AhoCorasickBuilder;
+
+#[no_mangle]
+#[allow(non_snake_case)]
+pub extern fn Java_App_hello(_env: JNIEnv, _class: JClass) {
+    println!("hello from rust!!!");
+}
+
+#[no_mangle]
+#[allow(non_snake_case)]
+pub extern fn Java_App_match(_env: JNIEnv, _class: JClass, keywords: JList, text: JString) -> Vec<usize> {
+    let ac = AhoCorasickBuilder::new()
+        .ascii_case_insensitive(true)
+        .build(keywords.collect::<Vec<_>>());
+    let mut matches = Vec<usize>::new();
+
+    for mat in ac.find_iter(text) {
+        matches.push(vec![mat.start(), mat.end()]);
```

Comment:
isso parece estar errado, já que o tipo de matches é `Vec<usize>` e não `Vec<Vec<uszie>>`. Além disso não ta faltando definir qual padrão que é o match com o `mat.pattern()` ? `(mat.pattern(), mat.start(), mat.end())`. E eu usaria tuplas de ambos os lados, mesmo que tenha que criar em java https://www.geeksforgeeks.org/triplet-class-in-java-tuples/

Reply:
O cargo compilou isso? roda um `cargo check` pra ver